provd/internal/services/telemetry/telemetry.go

```diff
@@ -0,0 +1,119 @@
+// Package telemetry implements the Telemetry gRPC service.
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	pb "github.com/canonical/ubuntu-desktop-provision/provd/protos"
+	"github.com/ubuntu/ubuntu-report/pkg/sysmetrics"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// Option is a functional options for tests.
```

Comment:
Nittiest of nitpicks:

```suggestion
// Option is a functional option for tests.
```

Reply:
Not at all! Good catch :smile: